AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/helper/IntentBuilder.java

```diff
@@ -135,16 +139,18 @@ public JInvocation getSuperPutExtraInvocation(TypeMirror elementType, JVar extra
 		if (elementType.getKind() == TypeKind.DECLARED) {
 			Elements elementUtils = environment.getProcessingEnvironment().getElementUtils();
 			TypeMirror parcelableType = elementUtils.getTypeElement(PARCELABLE).asType();
-			if (!typeUtils.isSubtype(elementType, parcelableType)) {
-				TypeMirror stringType = elementUtils.getTypeElement(STRING).asType();
-				if (!typeUtils.isSubtype(elementType, stringType)) {
-					extraParameterArg = cast(environment.getClasses().SERIALIZABLE, extraParameterArg);
-				}
-			} else {
+			if (typeUtils.isSubtype(elementType, parcelableType)) {
 				TypeMirror serializableType = elementUtils.getTypeElement(SERIALIZABLE).asType();
 				if (typeUtils.isSubtype(elementType, serializableType)) {
 					extraParameterArg = cast(environment.getClasses().PARCELABLE, extraParameterArg);
 				}
+			} else if (!BundleHelper.METHOD_SUFFIX_BY_TYPE_NAME.containsKey(elementType.toString()) && annotationHelper.isParcelType(elementType)) {
```

Comment:
This code is begging for a more unified approach.  For example, check out the way I did this 
in Parceler:
[Generators](https://github.com/johncarl81/parceler/blob/master/parceler/src/main/java/org/parceler/internal/Generators.java)
[Generators setup](https://github.com/johncarl81/parceler/blob/master/parceler/src/main/java/org/parceler/internal/ParcelerModule.java#L219-L258)
[Generators validation](https://github.com/johncarl81/parceler/blob/master/parceler/src/main/java/org/parceler/internal/ParcelableAnalysis.java#L575)
[Generators usage - generates read code](https://github.com/johncarl81/parceler/blob/master/parceler/src/main/java/org/parceler/internal/ParcelableGenerator.java#L284)
and in Transfuse:
[Bundle Property Repository](https://github.com/johncarl81/transfuse/blob/master/transfuse/src/main/java/org/androidtransfuse/analysis/repository/BundlePropertyBuilderRepository.java)
[Bundle Property Setup](https://github.com/johncarl81/transfuse/blob/master/transfuse/src/main/java/org/androidtransfuse/config/TransfuseAndroidModule.java#L219-L234)
[Bundle Property validation](https://github.com/johncarl81/transfuse/blob/master/transfuse/src/main/java/org/androidtransfuse/gen/variableBuilder/ExtraInjectionNodeBuilder.java#L93)
[Bundle Property usage - generates write code](https://github.com/johncarl81/transfuse/blob/master/transfuse/src/main/java/org/androidtransfuse/gen/IntentFactoryStrategyGenerator.java#L153)